chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ListingUploadController.java

```diff
@@ -0,0 +1,238 @@
+
+import javax.mail.MessagingException;
+import javax.persistence.EntityNotFoundException;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.ff4j.FF4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MaxUploadSizeExceededException;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import gov.healthit.chpl.FeatureList;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.IdListContainer;
+import gov.healthit.chpl.domain.ListingUpload;
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.exception.InvalidArgumentsException;
+import gov.healthit.chpl.exception.ObjectMissingValidationException;
+import gov.healthit.chpl.exception.ObjectsMissingValidationException;
+import gov.healthit.chpl.exception.ValidationException;
+import gov.healthit.chpl.upload.listing.ListingUploadManager;
+import gov.healthit.chpl.util.AuthUtil;
+import gov.healthit.chpl.util.EmailBuilder;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import lombok.extern.log4j.Log4j2;
+
+@Api(value = "listings")
+@RestController
+@RequestMapping("/listings")
+@Log4j2
+public class ListingUploadController {
+
+    @Value("${uploadErrorEmailRecipients}")
+    private String uploadErrorEmailRecipients;
+
+    @Value("${uploadErrorEmailSubject}")
+    private String uploadErrorEmailSubject;
+
+    private ListingUploadManager listingUploadManager;
+    private ErrorMessageUtil msgUtil;
+    private FF4j ff4j;
+    private Environment env;
+
+    @Autowired
+    public ListingUploadController(ListingUploadManager listingUploadManager,
+            FF4j ff4j, ErrorMessageUtil msgUtil, Environment env) {
+        this.listingUploadManager = listingUploadManager;
+        this.msgUtil = msgUtil;
+        this.ff4j = ff4j;
+        this.env = env;
+    }
+
+    @ApiOperation(value = "Get all uploaded listings to which the current user has access.",
+            notes = "Security Restrictions: User will be presented the uploaded listings that "
+                    + "they have access to according to ACB(s) and CHPL permissions.")
```

Comment:
```suggestion
                    + "they have access to according to ONC-ACB(s) and CHPL permissions.")
```

This is public facing, right? If so, I think we need to refer to ONC-ACBs with the ONC prefix